Redirections require a space to work

## Describe the bug

<!-- A clear and concise description of what the bug is. -->

Redirections require a space to work. i.e. to complete filenames.

## To reproduce

`cat <`<kbd>Tab</kbd> # gives no completion

`cat < `<kbd>Tab</kbd> # completes filenames as expected

Same thing with output redirections.

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

Redirections should complete files without having to enter a space after `>` or `<`.

## Versions (please complete the following information)

- [ ] Operating system name/distribution and version:
- [ ] bash version, `echo "$BASH_VERSION"`:
- [ ] bash-completion version, `(IFS=.; echo "${BASH_COMPLETION_VERSINFO[*]}")`: current master

## Additional context

This is a regression introduced in commit 38150dbd

## Debug trace

<!--
     See Troubleshooting section in README.md how to generate a debug trace,
     copy-paste it into a separate file and attach the file here.
-->

Thanks for the information. Maybe I miss something, but couldn't we just revert the change? Here, `redir` appears to contain a glob pattern:

https://github.com/scop/bash-completion/blob/b545d7d8598562edefd6ee4319eb257ee3609409/bash_completion#L1362

I think it is expected to match as a glob pattern in `${cur##$redir}`, so SC2295 would be a false positive.

- [ShellCheck: SC2295 – Expansions inside `${..}` need to be quoted separately, otherwise they will match as a pattern.](https://www.shellcheck.net/wiki/SC2295)

--- a/bash_completion
+++ b/bash_completion
@@ -1199,7 +1199,7 @@ _comp_delimited()
         delimiter=$1
         shift
     fi
-    [[ $cur == *$delimiter* ]] && prefix=${cur%"$delimiter"*}$delimiter
+    [[ $cur == *"$delimiter"* ]] && prefix=${cur%"$delimiter"*}$delimiter
 
     if [[ $deduplicate ]]; then
         # We could construct a -X pattern to feed to compgen, but that'd

--- a/bash_completion
+++ b/bash_completion
@@ -1382,7 +1382,8 @@ _comp_initialize()
                 esac
                 ;;
         esac
-        cur=${cur##"$redir"}
+        # shellcheck disable=SC2295 # redir is a pattern
+        cur=${cur##$redir}
         _comp_compgen_filedir "$xspec"
         return 1
     fi

--- a/test/t/unit/test_unit_initialize.py
+++ b/test/t/unit/test_unit_initialize.py
@@ -57,3 +57,10 @@ class TestUnitInitCompletion(TestUnitBase):
             bash, "cmd1 %s f" % redirect, cwd="shared/default"
         )
         assert "foo" in completion
+
+    @pytest.mark.parametrize("redirect", "> >> 2> < &>".split())
+    def test_redirect_3(self, bash, redirect):
+        completion = assert_complete(
+            bash, "cmd1 %sf" % redirect, cwd="shared/default"
+        )
+        assert "foo" in completion

